src/aiperf/dataset/composer/base.py

```diff
@@ -193,7 +193,7 @@ def _inject_context_prompts(self, conversations: list[Conversation]) -> None:
             if shared_system_prompt:
                 conversation.system_message = shared_system_prompt
                 self.trace(
-                    lambda: f"Set system_message on conversation {conversation.session_id}"
+                    lambda conv=conversation: f"Set system_message on conversation {conv.session_id}"
```

Comment:
I dont know if this is a necessary change. It adds 9 characters for no additional clarity.

Reply:
interesting. learned something today.